### PR TITLE
Adding snap support

### PR DESCRIPTION
**Pull request by barryprice**
Hi,

I've added a simple Snapcraft setup to the repo, which will automatically build a (sandboxed) version of the GUI app. Documentation added to the UNIX build page.

Currently this will build a QT4 version of the GUI app, but doesn't include the CLI utils (although they could certainly be added). I did have to disable UPnP support due to some assumptions the library makes about where it can open sockets. Otherwise though, it seems to work well.

I've also set up a (currently private/unpublished) registration for this in the Snap Store, which would  enable users on supported systems to be able to install the latest Dogecoin Core with a single command. Let me know if you'd be interested in supporting this, I'd be happy to transfer the registration over.

### CODE REVIEW

**Comment by langerhans** [comment]
Certainly interesting to have it easily running on those systems. Yet I'm wondering how much maintenance that would require. I'm not sure if any of the current devs is a main linux user so it would be tedious to keep this up to date 🤔 Any insights on that?

File: doc/build-unix.md [hunk 1]
```diff
@@ -222,3 +222,21 @@ In this case there is no dependency on Berkeley DB 4.8.
...
 call, not `getwork`.
 
+Snapcraft
+====
+
+You can also build Dogecoin Core with [Snapcraft](https://snapcraft.io/) on supported systems by pulling the source repo, and running 'snapcraft' within the root of the master branch.
+
+The resulting .snap file can then be installed locally via:
+
+    sudo snap install --dangerous <filename>
```
Comment (patricklodder) [inline doc/build-unix.md]: What is causing `--dangerous`?

**Comment by patricklodder** [comment]
I operate both a linux and an osx machine concurrently when I'm not on the road, so that shouldn't be an issue. Don't want to manually test each PR though, so can we perhaps run some basic set of tests through Travis?

File: doc/build-unix.md (hunk 1 above)
Comment (barryprice) [inline doc/build-unix.md]: Just the fact that it's being built locally, and so isn't coming from a trusted (cryptographically-signed) source. If there was interest in maintaining a compiled snap on the Snap Store, then this becomes a lot simpler.

**Comment by barryprice** [comment]
The existing YAML should continue to work for minor code updates.

If extra build dependencies are added, or the build system switches to something other than autotools, then the YAML would need to be updated to reflect that.

It looks [pretty straightforward to set up CI with Travis](https://snapcraft.io/docs/build-snaps/ci-integration), but that's not a tool I've previously used.

There's still work to be done for:
 - QT5: recent versions of qt5-base are built with -reduce-relocations, which means we need to explicitly build Dogecoin with -fPIC. This should be simple enough - I just haven't spent the time on it yet when QT4 worked fine
 - miniupnpc: We'd either need to patch the library or at least adjust the code that calls it to open its socket in TMPDIR or /tmp instead of /var/tmp - need to look more closely at this too
 - process-control: Dogecoin uses SetThreadPriority in a few places, which currently requires extra permissions to be manually granted (via the "sudo snap connect dogecoin-core:process-control core:process-control" line after it's installed). I wonder how much performance would suffer if those calls were made optional at compile-time, for the sake of a more manageable Snap. Longer-term, [there are plans to make this requirement less strict](https://bugs.launchpad.net/snappy/+bug/1577520).

**Comment by slightlyskepticalpotat** [comment]
Hello,

Is there still interest for this? The last update to this thread was in 2017, but snap adoption has only grown since then. I would be interested in working on a way to package the pre-built releases into a snap like [how Bitcoin does](https://github.com/bitcoin-core/packaging/blob/d4fa9f2174ab347b050703d25c78bab6ebd67e6c/snap/snapcraft.yaml).

**Comment by delbonis** [comment]
If you're targeting Ubuntu-family specifically then a PPA would be better in my somewhat-biased opinion.  Snap comes with some portability issues.  I may be wrong but are they even widely used outside the Ubuntu family?

**Comment by slightlyskepticalpotat** [comment]
Unfortunately, I was unable to find any official statistics of snap usage across distros. However, I found that the [core snap](https://snapcraft.io/core) is installed by default with snap and the installation statistics by distro (bottom right) is a good proxy for snap usage by distro. It looks like although the majority of snap users are Ubuntu as expected, some users running Debian, Manjaro, CentOS, and other Linux distros also use snap

**Comment by patricklodder** [comment]
merged with #1904

**Pull request closed without merging**